manager/orchestrator/task_reaper.go

```diff
@@ -75,13 +79,30 @@ func (tr *TaskReaper) Run() {
 				}] = struct{}{}
 				if len(tr.dirty) > maxDirty {
 					tr.tick()
+					if timer != nil {
+						timer.Stop()
+					}
+					timerCh = nil
```

Comment:
I think this is safe. I hesitated to do it because I wasn't sure if `Reset` uses the old channel or creates a new one. Do you know?